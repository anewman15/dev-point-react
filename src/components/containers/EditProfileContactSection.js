/* eslint-disable jsx-a11y/label-has-associated-control */

import PhoneInput from 'react-phone-input-2';
import 'react-phone-input-2/lib/style.css';

const EditProfileContactSection = () => (
  <div className="my-6">
    <h1 className="my-1 px-2 is-size-5 has-text-weight-bold">Contact Info</h1>
    <form>
      <div className="is-flex is-justify-content-start is-align-items-flex-end my-3">
        <label htmlFor="link-url" className="mx-2">
          Video Call Link
          <input className="input" type="url" name="link-url" placeholder="https://zoom.us/j/randomzoomidwithsomerandomcharsandnums" required />
        </label>
        <div className="control">
          <button type="submit" className="button is-primary">Add</button>
        </div>
      </div>
    </form>
    <form>
      <div className="is-flex is-justify-content-start is-align-items-flex-end my-3">
        <label htmlFor="phone-number" className="mx-2">
          Phone Number
          <PhoneInput country="us" value="+1 (123) 345 5678" />
        </label>
        <div className="control">
          <button type="submit" className="button is-primary">Add</button>
        </div>
      </div>
    </form>
  </div>
);

export default EditProfileContactSection;
