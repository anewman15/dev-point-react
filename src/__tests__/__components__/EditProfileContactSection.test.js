import { unmountComponentAtNode } from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import EditProfileContactSection from '../../components/containers/EditProfileContactSection';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('EditProfileContactSection', () => {
  it('renders the EditProfileContactSection component to the dom', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <EditProfileContactSection />
        </MemoryRouter>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
