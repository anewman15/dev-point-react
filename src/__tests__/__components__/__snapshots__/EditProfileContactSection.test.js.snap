// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfileContactSection renders the EditProfileContactSection component to the dom 1`] = `
<div
  className="my-6"
>
  <h1
    className="my-1 px-2 is-size-5 has-text-weight-bold"
  >
    Contact Info
  </h1>
  <form>
    <div
      className="is-flex is-justify-content-start is-align-items-flex-end my-3"
    >
      <label
        className="mx-2"
        htmlFor="link-url"
      >
        Video Call Link
        <input
          className="input"
          name="link-url"
          placeholder="https://zoom.us/j/randomzoomidwithsomerandomcharsandnums"
          required={true}
          type="url"
        />
      </label>
      <div
        className="control"
      >
        <button
          className="button is-primary"
          type="submit"
        >
          Add
        </button>
      </div>
    </div>
  </form>
  <form>
    <div
      className="is-flex is-justify-content-start is-align-items-flex-end my-3"
    >
      <label
        className="mx-2"
        htmlFor="phone-number"
      >
        Phone Number
        <div
          className=" react-tel-input"
          onKeyDown={[Function]}
          style={Object {}}
        >
          <div
            className="special-label"
          >
            Phone
          </div>
          <input
            className="input form-control"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onCopy={[Function]}
            onDoubleClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="1 (702) 123-4567"
            style={
              Object {
                "height": "40px",
              }
            }
            type="tel"
            value="+1 (123) 345-5678"
          />
          <div
            className=" flag-dropdown"
            style={Object {}}
          >
            <div
              aria-haspopup="listbox"
              className="selected-flag"
              onClick={[Function]}
              role="button"
              tabIndex="0"
              title="United States: + 1"
            >
              <div
                className="flag us"
              >
                <div
                  className="arrow"
                />
              </div>
            </div>
          </div>
        </div>
      </label>
      <div
        className="control"
      >
        <button
          className="button is-primary"
          type="submit"
        >
          Add
        </button>
      </div>
    </div>
  </form>
</div>
`;
